apps/condo/domains/news/utils/serverSchema/recipientsCounterUtils.js

```diff
@@ -0,0 +1,99 @@
+const compact = require('lodash/compact')
+const filter = require('lodash/filter')
+const find = require('lodash/find')
+const get = require('lodash/get')
+const identity = require('lodash/identity')
+const pick = require('lodash/pick')
+const pickBy = require('lodash/pickBy')
+
+const { loadListByChunks } = require('@condo/domains/common/utils/serverSchema')
+const { queryFindResidentsByOrganizationAndScopes } = require('@condo/domains/news/utils/accessSchema')
+const { Property } = require('@condo/domains/property/utils/serverSchema')
+const { Resident } = require('@condo/domains/resident/utils/serverSchema')
+
+const getUnitsFromProperty = (property) => (
+    property?.map?.sections?.reduce((acc, section) => ([
+        ...acc,
+        ...getUnitsFromSection(section),
+    ]), []) || []
+)
+
+const getUnitsFromSection = (section) => section.floors.flatMap(floor => floor.units.map(unit => ({
+    unitName: unit.label,
+    unitType: unit.unitType,
+})))
+
+async function getUnitsData (context, organizationId, newsItemScopes) {
```

Comment:
It looks too heavy. Why not to use count query?

Reply:
Or what do you mean?

Reply:
Ok, the code was refactored to decrease memory usage.